Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -380,77 +395,92 @@ public void JumpCheck()
 				jumpColCheckTimer = 0.0f;
 				//actor.GetAnimator().SetBool("isSliding", false);
 			}
-			else if ( jumpColCheckTimer > jumpColCheckTime && isOnGround && !IsInState( ActorStates.Rolling ) )
-			{
-				if ( !IsInState( ActorStates.Grounded ) )
-				{
-					// if !launching
-					ChangeState( ActorStates.Grounded );
-					stopMoveTimer = 0f;
-					//GainControl();
-				}
-
-				actor.animator.SetBool( "isJumping", false );
-				//actor.GetAnimator().SetBool("isSliding", false);
-				lateJumpTimer = 0.0f;
-			}
+			//else if ( jumpColCheckTimer > jumpColCheckTime && isOnGround && !IsInState( ActorStates.Rolling ) )
+			//{
+			//	if ( !IsInState( ActorStates.Grounded ) )
+			//	{
+			//		// if !launching
+			//		ChangeState( ActorStates.Grounded );
+			//		stopMoveTimer = 0f;
+			//		//GainControl();
+			//	}
+
+			//	actor.animator.SetBool( "isJumping", false );
+			//	//actor.GetAnimator().SetBool("isSliding", false);
+			//	lateJumpTimer = 0.0f;
+			//}
 		}
-		else if ( !IsInState( ActorStates.Jumping ) && !IsInState( ActorStates.Rolling ) ) // if not currently being launched
-		{
-			if ( actor.animator != null )
-			{
-				actor.animator.SetBool( "isJumping", true );
-				//actor.GetAnimator().SetBool("isSliding", false);
-			}
+		//else if ( !IsInState( ActorStates.Jumping ) && !IsInState( ActorStates.Rolling ) ) // if not currently being launched
+		//{
+		//	Debug.Log( "Not on Ground" );
 
-			ChangeState( ActorStates.Jumping );
-		}
+		//	if ( actor.animator != null )
+		//	{
+		//		actor.animator.SetBool( "isJumping", true );
+		//		//actor.GetAnimator().SetBool("isSliding", false);
+		//	}
+
+		//	ChangeState( ActorStates.Jumping );
+		//}
 
 		if ( !IsInState( ActorStates.Grounded ) )
 		{
 			jumpColCheckTimer += Time.deltaTime;
 		}
-
-		lateJumpTimer += Time.deltaTime;
-		rigidbody.velocity = Vector3.ClampMagnitude( rigidbody.velocity, maxSpeed );
 	}
 
-
-
-	public void ModelControl()
+	/**
+	 * @brief Orients the model to face the direction of movement
+	 * 
+	 * @details
+	 *     Orientation is reactive to velocity, meaning the
+	 *     model will face the direction of movement, rather
+	 *     than the input direction.
+	 */
+	public void OrientSelf()
```

Comment:
Yeah, I've got a fix. I changed `ActorPhysics` to rotate the root transform rather than the model's transform, but there were other places in code that were still rotation the model. I cleaned those up and that fixed the problem.